libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -541,6 +603,52 @@ void AP_ESC_Telem::update()
                 _last_telem_log_ms[i] = telemdata.last_update_ms;
                 _last_rpm_log_us[i] = rpmdata.last_update_us;
             }
+
+#ifdef HAL_WITH_BIDIR_DSHOT
+            // Write an EDTv2 status message, if there is any update
+            uint16_t edt2_status = telemdata.edt2_status;
+            if (EDT2_HAS_NEW_DATA(edt2_status)) {
+                // An EDT2 status message is:
+                //   id starts from 0
+                //   alert, warning, error are flags (so 0 or 1)
+                //   stress is in [0; 15]
+                const struct log_Edt2 pkt_edt2{
+                    LOG_PACKET_HEADER_INIT(uint8_t(LOG_EDT2_MSG)),
+                    time_us     : now_us64,
+                    instance    : i,
+                    alert       : EDT2_ALERT_BIT_FROM_STATUS(edt2_status),
+                    warning     : EDT2_WARNING_BIT_FROM_STATUS(edt2_status),
+                    error       : EDT2_ERROR_BIT_FROM_STATUS(edt2_status),
```

Comment:
These can be combined into a single uint8 flags field with our new logging bitmask documentation the log review tools should be able to pull out the data.

This is a example for PID:

https://github.com/ArduPilot/ardupilot/blob/e9d065c1cc011ab879f31285991aa6c6e57d102c/libraries/AP_Logger/LogFile.cpp#L478-L497

The enum is then referenced here:

https://github.com/ArduPilot/ardupilot/blob/e9d065c1cc011ab879f31285991aa6c6e57d102c/libraries/AP_Logger/LogStructure.h#L930-L931

Reply:
It should work for multi bit values too. So you could do the max_stress in the same way, but it might be a little odd having that under "flags".

Reply:
This mapping is not written anywhere in the logs, right? So anything except for Mission Planner, which has these mappings compiled-in, is unlikely to interpret that correctly. This will be of no use to me then.

Reply:
The data is still there, it is just not broken out in to fields nicely. Your right, we should support it in other tools as well. For me it makes little difference as I already don't know what "error" "warning" and "alert" really mean so we might as well save the log space.

Reply:
I am not sure whether the savings will be large, as all these `uint8_t`s take 40 bits compared to the timestamp of 64 bits. And given the message rate of EDT2, which is roughly once per second, this is a tiny fraction of everything else logged, I think.

(I actually tried to log the entire status as a byte in the early stages of this development, and it was really hard to understand what is happening from this byte, the mental bit juggling involved was somewhat too much for me)

Overall, to me it makes more sense to keep this piece in its current form now, and convert it to a more compressed form once the tools are ready.

Reply:
I agree that having the four bit `max_stress` in there makes it too confusing for manual review. But the three single bit values are not too bad.

My point is really that you already need some extra knowledge that it not present in the log. Without knowledge of what the fields actual mean the information is not useful. If I get a "error" do I need to replace my ESC? Maybe if I get a "warning" I should land immediately? This extra information is provided by the log documentation, in this case it might be ESC dependent, but the log documentation can tell me to go and look at the ESC documentation. So if I have to look at the documentation anyway we can document the bitmask too.

The documentation is also used for the wiki, going back to the PID example:
https://ardupilot.org/copter/docs/logmessages.html#pidp

So although it would be nice if the tools did it automatically, that they don't does not put the information completely out of reach.